src/cript/api/api.py

```diff
@@ -38,52 +38,20 @@ def _get_global_cached_api():
     return _global_cached_api
 
 
-def _prepare_host(host: str) -> str:
-    """
-    prepares the host and gets it ready to be used within the api client
-
-    1. removes any trailing spaces from the left or right side
-    1. removes "/" from the end so that it is always uniform
-    1. adds "/api", so all queries are sent directly to the API
-
-    Parameters
-    ----------
-    host: str
-        api host
-
-    Returns
-    -------
-    host: str
-    """
-    # strip any empty spaces on left or right
-    host = host.strip()
-
-    # strip ending slash to make host always uniform
-    host = host.rstrip("/")
-
-    host = f"{host}/api/v1"
-
-    # if host is using unsafe "http://" then give a warning
-    if host.startswith("http://"):
-        warnings.warn("HTTP is an unsafe protocol please consider using HTTPS.")
-
-    if not host.startswith("http"):
-        raise InvalidHostError("The host must start with http or https")
-
-    return host
-
-
 class API:
     """
     ## Definition
     API Client class to communicate with the CRIPT API
     """
 
-    _host: str = ""
+    _raw_host: str = ""
+    _https_warning_sent: bool = False
```

Comment:
why do we need to know if we gave a warning already or not?

I think if the user sees the warning multiple times it is okay, there are other libraries where they give a warning multiple times

I think if we had it like before with prepare host a lot of these could just be encapsulated into there safely and neatly in one place and warning would only get triggered once because it would only hit prepare host on `__init__`

Reply:
We can get rid of it.
I just thought issueing many warning might be annoying.